std/traits.d

```diff
@@ -8087,3 +8089,124 @@ enum isCopyable(S) = is(typeof(
     static assert(isCopyable!int);
     static assert(isCopyable!(int[]));
 }
+
+private template TypeOf(T...)
+    if (T.length == 1 && (is(T[0]) || is(typeof(T[0]))))
+{
+    static if (is(T[0]))
+        alias TypeOf = T[0];
+    else static if (is(typeof(T[0])))
+        alias TypeOf = typeof(T[0]);
+}
+
+@safe unittest
+{
+    assert(is(TypeOf!3 == int));
+    assert(is(TypeOf!int == int));
+    assert(is(TypeOf!([]) == void[]));
+    assert(!is(TypeOf!(mixin(__MODULE__))));
+    assert(is(TypeOf!TypeOf));
+}
+
+/**
+ * Get a list of the overloads for the given callable.
+ *
+ * Params:
+ *     func = The callable whose overload set to return.
+ * Returns:
+ *     A tuple of the overloads for the given callable.
+ */
+template getOverloads(func...)
+    if (func.length == 1 && isCallable!func)
+{
+    enum isModuleOrPackage(alias s) = !__traits(compiles, { alias _ = typeof(s); });
+    alias Parent = AliasSeq!(__traits(parent, func))[0];
+
+    static if (is(Parent) || isModuleOrPackage!Parent)
+        alias getOverloads = AliasSeq!(__traits(getOverloads, Parent, __traits(identifier, func)));
+    else
+        alias getOverloads = func;
```

Comment:
Perhaps add a comment with an example of when the `else` branch is triggered?

Reply:
Done.